app/models/printer.rb

```diff
@@ -19,4 +19,14 @@ class Printer < ApplicationRecord
       filters
     end
   end
+
+  after_save do
+    existing_printers = `lpstat -a`.split("\n")
+    existing_printers.map! { |printer| printer.split(' ').first }
+    unless existing_printers.include? name
+      if Rails.configuration.auto_create_printer_in_cupsd
+        `sudo lpadmin -p #{name} -v socket://#{name}.internal.sanger.ac.uk -E`
```

Comment:
I'm thinking now about using name. We dont check anything in its contents, but if it is a user accessible field, I can create a printer with any name I want.
The class Printer doesn't do any checks in the name values.  What would happen if I create a printer like this:

```Printer.create(name: "test; echo 'hi';");```

Would I be able to run the command (echo hi) with user root?
How could we solve this so users cannot inject code inside the name?

Reply:
Very good with having it in a after_save. An improvement could be to have this code in its own method
```
def update_printer_in_cups
...
end
```

and then defining the callback with the reference to the method:
```
after_save :update_printer_in_cups
```

This way we can call this method manually if we ever need to.